src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -239,10 +255,85 @@ ProjectItemInstance projectReference
                         $"{setConfigurationString};{setPlatformString};{setTargetFrameworkString}").ToImmutableDictionary();
                 }
             }
-
             return new GlobalPropertyPartsForMSBuildTask(newProperties, defaultParts.AdditionalProperties, newUndefineProperties);
         }
 
+        private static string GetCompatiblePlatform2(ProjectItemInstance projectReference){
```

Comment:
What's this "2" method? Shouldn't you be using `PlatformNetotiation`?

Reply:
this was prior to extracting it to a shared location. removed